src/middlewares/one-of.ts

```diff
@@ -9,16 +9,19 @@ const dummyItem: ContextItem = { async run() {} };
 
 export type OneOfCustomMessageBuilder = (options: { req: Request }) => any;
 
+export type OneOfErrorType = 'grouped' | 'leastErroredOnly' | 'legacy';
+
+export interface OneOfOptions {
+  message?: OneOfCustomMessageBuilder | any;
```

Comment:
`| any` flattens the type to `any`, which is why there was 2 `oneOf` signatures.
Do you want to do something like `type OneOfOptions = { message: something } | { message: any }`?

Reply:
That seems to still erase the type. Maybe the two signatures approach is the only way?

Reply:
Probably? That'd look good to me.